tests/api/test_api.py

```diff
@@ -9,3 +9,75 @@ def test_get_output_values_prop(self):
         c.add_node(A)
         result = c.run(inputs={A: [1]}, num_trials=2)
         assert result == c.output_values == [np.array([1])]
+
+
+    # test whether xor model created as autodiff composition learns properly
+    @pytest.mark.parametrize("mode", ['Python',
+                                      pytest.param('LLVMRun', marks=pytest.mark.llvm),
```

Comment:
What about `LLVMExec`? does it work? should it be tested as well?

Reply:
Learning should only work using `LLVMRun` at present - it could theoretically be possible to support `LLVMExec` if a writeback step was added after compiled execute.

Reply:
sounds good. `LLVMExec` support can be added later for easier debugging, but I don't expect anyone to need it.